network/src/commonMain/kotlin/dev/icerock/moko/network/features/HttpRetrySendFeature.kt

```diff
@@ -0,0 +1,76 @@
+package dev.icerock.moko.network.features
+
+import io.ktor.client.HttpClient
+import io.ktor.client.features.HttpClientFeature
+import io.ktor.client.request.HttpRequestBuilder
+import io.ktor.client.request.HttpSendPipeline
+import io.ktor.client.statement.HttpStatement
+import io.ktor.util.AttributeKey
+import io.ktor.utils.io.errors.IOException
+import kotlinx.coroutines.delay
+
+/**
+@param onGetDelay the callback for calculation a delay between failed request and next request
+@param maxAmountRetrying the max amounts retrying requests
+@param onConditionRetrying the callback for condition a retrying request. By default it check `e is IOException `
```

Comment:
naming invalid. `onGetDelay` called as callback, but it not "callback" - it behavior function. i suggest `delayGetter: (lastDelayInMillisecond: Long, timeRetrying: Int) -> Long` naming.
for `onConditionRetrying` - `isShouldRetryRequest(e: Throwable): Boolean`